backend/api/dataComplianceApi.test.js

```diff
@@ -6,38 +6,75 @@ const context = { some: 'context' }
 
 describe('data compliance api', () => {
   describe('get offender retention reasons', () => {
-    const response = { some: 'response' }
+    const responseBody = { some: 'response' }
     let actual
 
-    beforeEach(() => {
-      client.get = jest.fn().mockReturnValue({ then: () => response })
-      actual = dataComplianceApi.getOffenderRetentionReasons(context)
+    beforeEach(async () => {
+      client.get = jest.fn().mockResolvedValue({ body: responseBody })
+      actual = await dataComplianceApi.getOffenderRetentionReasons(context)
     })
 
     it('should call the correct endpoint', () => {
       expect(client.get).toBeCalledWith(context, '/retention/offenders/retention-reasons')
     })
 
-    it('should return response from endpoint', () => {
-      expect(actual).toEqual(response)
+    it('should return response from endpoint', async () => {
+      expect(actual).toEqual(responseBody)
     })
   })
 
   describe('get offender retention record', () => {
-    const response = { some: 'response' }
+    const responseBody = { etag: '"0"', some: 'response' }
     let actual
 
-    beforeEach(() => {
-      client.get = jest.fn().mockReturnValue({ then: () => response })
-      actual = dataComplianceApi.getOffenderRetentionRecord(context, 'A1234BC')
+    beforeEach(async () => {
+      client.get = jest.fn().mockResolvedValue({ headers: { etag: '"0"' }, body: { some: 'response' } })
+      actual = await dataComplianceApi.getOffenderRetentionRecord(context, 'A1234BC')
     })
 
     it('should call the correct endpoint', () => {
       expect(client.get).toBeCalledWith(context, '/retention/offenders/A1234BC')
     })
 
-    it('should return response from endpoint', () => {
-      expect(actual).toEqual(response)
+    it('handles jetty bug where gzip encoding causes etag suffix', async () => {
```

Comment:
Was encountering this with the Wiremock tests so have handled it to be safe https://github.com/eclipse/jetty.project/issues/1416